tsdb/chunkenc/histogram.go

```diff
@@ -1121,3 +1121,71 @@ func resize[T any](items []T, n int) []T {
 	}
 	return items[:n]
 }
+
+type histogramStatsIterator struct {
+	Iterator
+	hReader  *histogram.Histogram
+	fhReader *histogram.FloatHistogram
+}
+
+func NewHistogramStatsIterator(it Iterator) Iterator {
+	return histogramStatsIterator{
+		Iterator: it,
+		hReader:  &histogram.Histogram{},
+		fhReader: &histogram.FloatHistogram{},
+	}
+}
+
+func (f histogramStatsIterator) AtHistogram(h *histogram.Histogram) (int64, *histogram.Histogram) {
+	var t int64
+	t, f.hReader = f.Iterator.AtHistogram(f.hReader)
+	if value.IsStaleNaN(f.hReader.Sum) {
+		return t, &histogram.Histogram{Sum: f.hReader.Sum}
+	}
+
+	if h == nil {
+		return t, &histogram.Histogram{
+			CounterResetHint: f.hReader.CounterResetHint,
+			Count:            f.hReader.Count,
+			ZeroCount:        f.hReader.ZeroCount,
+			Sum:              f.hReader.Sum,
+			ZeroThreshold:    f.hReader.ZeroThreshold,
+			Schema:           f.hReader.Schema,
```

Comment:
Since the schema is only for buckets, I think we can set this to any number without doing harm. If we set it consistently to the same number (e.g. 0), we avoid the CopyToSchema step.

Reply:
I removed both the zero threshold and schema fields.